src/citrine/informatics/predictors/chemical_formula_featurizer.py

```diff
@@ -152,5 +153,23 @@ def __init__(self,
         self.excludes = excludes if excludes is not None else []
         self.powers = powers if powers is not None else [1]
 
+    @property
+    def powers(self) -> List[int]:
+        """The list of powers when computing generalized weighted means of element properties."""
+        warn("The type of 'powers' will change to a list of floats in v4.0.0. To retrieve them as "
+             "floats now, use 'powers_float'.")
+        return [int(p) for p in self._powers]
+
+    @powers.setter
+    def powers(self, value: List[Union[int, float]]):
+        self._powers = value
```

Comment:
```suggestion
        self._powers = [float(p) for p in value]
```
If we're being particular about int v. float, then the above setter would cause powers_float to potentially return a list of ints, breaking the API.

Reply:
Nope. The serde library converts them automatically.

Reply:
So then you don't actually accept int to the setter, and your TypeHint is incorrect.

Reply:
Concern withdrawn; the `properties` module handles casting the result.